pkg/segment/query/metadata/metautils/metacheckers.go

```diff
@@ -106,7 +106,7 @@ func doesUintPassRangeFilter(op utils.FilterOperator, lookupValue uint64, minVal
 	case utils.Equals:
 		return lookupValue >= minVal && lookupValue <= maxVal
 	case utils.NotEquals:
-		if lookupValue == minVal || lookupValue == maxVal {
+		if minVal == maxVal && lookupValue == minVal {
```

Comment:
I think your change is correct, but not as restrictive as it could be. I think `utils.NotEquals` should be the same as `not utils.Equals`, which is
```
not (lookupValue >= minVal && lookupValue <= maxVal)
```
which is
```
lookupValue < minVal || lookupValue > maxVal
```
Am I missing something?

Reply:
Also, can you check the other inequalities? They look odd to me but I'm not quite convinced if they're right/wrong. There's a basic unit test for this function, but it would be good to add some more tests for it to catch edge cases like the one you found

Reply:
Wouldn't this condition `lookupValue < minVal || lookupValue > maxVal` give incorrect results for the case where lookupValue is 300, minVal is 200 and maxVal is 500. It would give `false` for that block even though there are records satisfying this condition.

Reply:
Ah, I think I was misunderstanding exactly how this function is supposed to behave. Now I think this is the expected behavior:

Returns true if there exists an `actualValue` with `minValue <= actualValue <= maxValue` such that `actualValue op lookupValue` is true

Is that what these functions are supposed to do? Can you please add a comment to the code explaining what the function does, and then I'll approve this PR

Reply:
Yes that is correct.